cmd/analyzer/parse_args.go

```diff
@@ -249,39 +259,50 @@ func minMaxValidity(minPort, maxPort int64, minPortName, maxPortName string) err
 }
 
 func validRangeConnectionExplainMode(args *InArgs) error {
-	err := minMaxValidity(*args.QSrcMinPort, *args.QSrcMaxPort, QSrcMinPort, QSrcMaxPort)
+	err := minMaxValidity(*args.ESrcMinPort, *args.ESrcMaxPort, ESrcMinPort, ESrcMaxPort)
 	if err != nil {
 		return err
 	}
-	err = minMaxValidity(*args.QDstMinPort, *args.QDstMaxPort, QDstMinPort, QDstMaxPort)
+	err = minMaxValidity(*args.EDstMinPort, *args.EDstMaxPort, EDstMinPort, EDstMaxPort)
 	if err != nil {
 		return err
 	}
 
-	if !PortInRange(*args.QSrcMinPort) || !PortInRange(*args.QSrcMaxPort) ||
-		!PortInRange(*args.QDstMinPort) || !PortInRange(*args.QDstMaxPort) {
+	if !PortInRange(*args.ESrcMinPort) || !PortInRange(*args.ESrcMaxPort) ||
+		!PortInRange(*args.EDstMinPort) || !PortInRange(*args.EDstMaxPort) {
 		return fmt.Errorf("%s, %s, %s and %s must be in ranges [%d, %d]",
-			QSrcMinPort, QSrcMaxPort, QDstMinPort, QDstMaxPort, common.MinPort, common.MaxPort)
+			ESrcMinPort, ESrcMaxPort, EDstMinPort, EDstMaxPort, common.MinPort, common.MaxPort)
 	}
 
 	return nil
 }
 
 func invalidArgsExplainMode(args *InArgs, flagset *flag.FlagSet) error {
-	if *args.AnalysisType != explainMode && wereExplainParamsSpecified(flagset) {
-		return fmt.Errorf("%s, %s, %s, %s and %s can be specified only when analysis-type is %s",
-			QProtocol, QSrcMinPort, QSrcMaxPort, QDstMinPort, QDstMaxPort, explainMode)
+	if *args.AnalysisType != explainMode {
+		if wereExplainParamsSpecified(flagset, []string{ESrc, EDst, EProtocol, ESrcMinPort, ESrcMaxPort, EDstMinPort, EDstMaxPort, explainMode}) {
+			return fmt.Errorf("%s, %s, %s, %s, %s, %s and %s can be specified only when analysis-type is %s",
+				ESrc, EDst, EProtocol, ESrcMinPort, ESrcMaxPort, EDstMinPort, EDstMaxPort, explainMode)
+		}
+		return nil
 	}
 
-	if *args.AnalysisType != explainMode {
+	if *args.ESrc == "" || *args.EDst == "" {
+		return fmt.Errorf("please specify %s and %s network_interface / external ip you want to explain connectivity for", ESrc, EDst)
+	}
+
+	if *args.EProtocol == "" {
```

Comment:
In `ExplainMode` `Protocol` must be specified. So you should simple issue a corresponding error msg here, regardless of `wereExplainParamsSpecified(...)`